src/aliases/api/handlers/handlers.go

```diff
@@ -0,0 +1,297 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"sync"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	jsonutils "github.com/consensys/quorum-key-manager/pkg/json"
+	"github.com/consensys/quorum-key-manager/src/aliases/api/types"
+	aliasent "github.com/consensys/quorum-key-manager/src/aliases/entities"
+	infrahttp "github.com/consensys/quorum-key-manager/src/infra/http"
+	"github.com/gorilla/mux"
+)
+
+type AliasHandler struct {
+	alias aliasent.AliasBackend
+}
+
+func NewAliasHandler(backend aliasent.AliasBackend) *AliasHandler {
+	h := AliasHandler{
+		alias: backend,
+	}
+
+	return &h
+}
+
+func (h *AliasHandler) Register(r *mux.Router) {
+	regRoute := r.PathPrefix("/registries/{registry_name}").Subrouter()
+	regRoute.HandleFunc("", h.deleteRegistry).Methods(http.MethodDelete)
+
+	alRoute := regRoute.PathPrefix("/aliases").Subrouter()
+	alRoute.HandleFunc("", h.listAliases).Methods(http.MethodGet)
+	alRoute.HandleFunc("/{alias_key}", h.createAlias).Methods(http.MethodPost)
+	alRoute.HandleFunc("/{alias_key}", h.getAlias).Methods(http.MethodGet)
+	alRoute.HandleFunc("/{alias_key}", h.updateAlias).Methods(http.MethodPut)
+	alRoute.HandleFunc("/{alias_key}", h.deleteAlias).Methods(http.MethodDelete)
+}
+
+// @Summary Delete a registry
+// @Description Delete a registry and all its keys
+// @Tags registry
+// @Param registry_name path string true "registry identifier"
+// @Success 204 {object} types.DeleteRegistryResponse "Registry data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 404 {object} ErrorResponse "Registry not found"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /registries/{registry_name} [delete]
+func (h *AliasHandler) deleteRegistry(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+
+	err := validatePathVars(regName)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	err = h.alias.DeleteRegistry(r.Context(), aliasent.RegistryName(regName))
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// @Summary Creates an alias
+// @Description Create an alias of a key in a dedicated alias registry
+// @Tags alias
+// @Accept json
+// @Produce json
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Param request body types.AliasRequest true "Create Alias Request"
+// @Success 200 {object} types.AliasResponse "Alias data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
```

Comment:
You are missing 404

Reply:
There is no 404 as the registry will be created if it doesn't exist yet (it is just an ID that is part of the key).

Reply:
I see, resolving